### PR TITLE
Add option to render recording locally using "maestro record --local"

### PR DESCRIPTION
**Pull request by Leland-Takamine**
Uses Jetbrain's [Skiko](https://github.com/JetBrains/skiko) to render video locally. Skiko is a Kotlin library that provides bindings to [Skia](https://skia.org/) which is the rendering engine for Google Chrome and ChromeOS, Android, Flutter, and others.

**Why Skia?**
* It is a battle-tested piece of software (Google Chrome and ChromeOS, Android, Flutter)
* It can run on Mac, Linux, or Windows and x86 + ARM
* There is a well-maintained JVM bindings library (Skiko)
* It doesn't require anything additional to be installed by users
* It has modern rendering capabilities like shadows and color emoji rendering
* It can run headless (without a browser or display)

The other alternatives I looked into fell short in one way or another (eg. AWT, JavaFX). Skia checked all the boxes.

Example of locally rendered video:

https://github.com/user-attachments/assets/6d3d715d-a1cd-4341-b088-e6ed3c09018b

### CODE REVIEW

File: maestro-cli/src/main/java/maestro/cli/command/RecordCommand.kt [hunk 1]
```diff
@@ -111,87 +123,21 @@ class RecordCommand : Callable<Int> {
...
-
-                Thread.sleep(2000)
-            }
+            val localOutputFile = outputFile ?: path.resolve("maestro-recording.mp4").toFile()
+            val videoRenderer = if (local) LocalVideoRenderer(
+                frameRenderer = SkiaFrameRenderer(),
+                outputFile = localOutputFile,
+                outputFPS = 25,
+                outputWidthPx = 1920,
+                outputHeightPx = 1080,
```
Comment (amanjeetsingh150) [inline maestro-cli/src/main/java/maestro/cli/command/RecordCommand.kt:134]: Are these the same specs for the video we used to record with remote renderer?

File: maestro-cli/src/main/java/maestro/cli/graphics/SkiaFrameRenderer.kt [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+import org.jetbrains.skia.Surface
+import org.jetbrains.skia.paragraph.FontCollection
+import org.jetbrains.skia.paragraph.ParagraphBuilder
+import org.jetbrains.skia.paragraph.ParagraphStyle
+import org.jetbrains.skia.paragraph.TextStyle
+import org.jetbrains.skiko.toImage
+import java.awt.image.BufferedImage
+import javax.imageio.ImageIO
+
+class SkiaFrameRenderer : FrameRenderer {
```
Comment (amanjeetsingh150) [inline maestro-cli/src/main/java/maestro/cli/graphics/SkiaFrameRenderer.kt:18]: I am thinking about what would be the best way to get this from beta to prod (stable):

1. Recommending to use this one to everyone who raised a Github issue for the remote renderer
2. Announcing this in the contributor's lounge or just the announcement channel to see if it works for everyone.

WDYT?

**Review by amanjeetsingh150 — APPROVED** [review]

File: maestro-cli/src/main/java/maestro/cli/graphics/LocalVideoRenderer.kt [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+                    var nextFrame: PictureWithMetadata? = grab.nativeFrameWithMetadata
+                    (0..outputFrameCount).forEach { frameIndex ->
+                        val currentTimestampSeconds = frameIndex.toDouble() / outputFPS
+
+                        // !! Due to smart cast limitation: https://youtrack.jetbrains.com/issue/KT-7186
+                        @Suppress("UNNECESSARY_NOT_NULL_ASSERTION")
+                        while (nextFrame != null && nextFrame!!.timestamp <= currentTimestampSeconds) {
+                            curFrame = nextFrame!!
+                            nextFrame = grab.nativeFrameWithMetadata
+                        }
```
Comment (bartekpacia) [inline maestro-cli/src/main/java/maestro/cli/graphics/LocalVideoRenderer.kt:57]: looks like this bug is fixed in Kotlin 2.0

([currently Maestro is at 1.8.22](https://github.com/mobile-dev-inc/maestro/blob/c67756875ccca2567ed6df38f488f5dd2919ca63/gradle/libs.versions.toml#L36))

File: maestro-cli/src/main/java/maestro/cli/command/RecordCommand.kt (hunk 1 above)
Comment (Leland-Takamine) [reply maestro-cli/src/main/java/maestro/cli/command/RecordCommand.kt:134]: Remote recorder was 30 fps but I think 25 if sufficient

File: maestro-cli/src/main/java/maestro/cli/graphics/SkiaFrameRenderer.kt (hunk 2 above)
Comment (Leland-Takamine) [reply maestro-cli/src/main/java/maestro/cli/graphics/SkiaFrameRenderer.kt:18]: Yeah I think we can announce in slack and have folks try it out. Lot's of users have asked for this

File: maestro-cli/src/main/java/maestro/cli/graphics/LocalVideoRenderer.kt (hunk 3 above)
Comment (Leland-Takamine) [reply maestro-cli/src/main/java/maestro/cli/graphics/LocalVideoRenderer.kt:57]: Ah nice good to know!

**Pull request merged**